src/main/webapp/meal-page-script.js

```diff
@@ -95,10 +98,39 @@ function createMap() {
       }
     })
     .then(() => {
-      addRestaurants(map);
+      const request = {
+        location: map.getCenter(),
+        radius: "1500",
+        type: ["restaurant", "cafe", "meal_takeaway", "bar", "bakery"],
+        query: type,
+      };
+      const service = new google.maps.places.PlacesService(map);
+      getSearchPromise(service, request, map).then((results) => {
```

Comment:
If you `return` the `getSearchPromise` result here, you'll be able to handle its failures afterwards.

Let's `return getSearchPromise` here and add a `.catch(err => console.log(`Some descriptive message here: ${err}`))` in line 111.